modules/renter/downloadcache.go

```diff
@@ -14,21 +14,33 @@ import (
 // TODO this won't be necessary anymore once we have partial downloads.
 func (udc *unfinishedDownloadChunk) addChunkToCache(data []byte) {
 	if udc.download.staticDestinationType == destinationTypeSeekStream {
-		// We only cache streaming chunks since browsers and media players tend to only request a few kib at once when streaming data. That way we can prevent scheduling the same chunk for download over and over.
+		// We only cache streaming chunks since browsers and media players tend
+		// to only request a few kib at once when streaming data. That way we can
+		// prevent scheduling the same chunk for download over and over.
 		return
 	}
 	udc.cacheMu.Lock()
+	defer udc.cacheMu.Unlock()
+
 	// Prune cache if necessary.
-	// TODO insteado of deleting a 'random' key, delete the
-	// least-recently-accessed element of the cache.
-	for key := range udc.chunkCache {
-		if len(udc.chunkCache) < downloadCacheSize {
-			break
+	if len(udc.chunkCache) >= downloadCacheSize {
```

Comment:
should be a loop `for len(udc.chunkCache) >= downloadCacheSize {`

Reply:
I don't think it needs to be since we never add more than 1 element at once and we also check the cacheSize every time we add a single element.